### ISSUE TITLE
Incorrect match type for nonsynonymous mutations

### ISSUE THREAD

**Issue by BertBog** [rastanton/GAMMA#7]
Dear,

I am testing GAMMA to detect virulence genes in assembled contigs from bacteria.
On my dataset I get many results with match type 'Truncation' which should be 'Mutant' instead.

I have attached a minimal example with the input contig (*contig_in.fasta*) and database sequence (*lukE_6.fasta*).
[example_gamma.zip](https://github.com/rastanton/GAMMA/files/8261172/example_gamma.zip)

There is a single (nucleotide) mutation when aligning the reference sequence to the contig (position 1839 in the contig), where a T is present in the contig and a G is present in the reference gene.

When translating these sequences this results in an amino acid mutation at position 145 in the amino acid (Cys -> Gly), which is not a nonsense mutation. So GAMMA should report 'Mutant' instead of 'Truncation' I think. In fact, I have not observed any 'Mutant' match type in my dataset, so I assume this might be a bug?

GAMMA command:
`GAMMA.py contig_in.fasta lukE_6.fasta gamma_out`

I have included the GAMMA output in the ZIP archive.

Best regards,
Bert

**Comment by rastanton** [comment]
Hi Bert,

Thank you for the message. I tried running your contig and gene fasta on my own system and got a different result, which looks like what you said you should have gotten. Here's the output O just generated:

lukE:6:BA000033.2	contig_in	1406	2342	Mutant	G145C,	1	1	1	0.9968	0.9989	1	936	936	+

It looks like that is what you said it should have been, and I agree that the "gamma_out.gamma" file you sent is incorrect--it says that there are no coding mutations yet lists the match type as a truncation--that does occur in cases where a gene in the database itself has a truncation/stop codon in the middle of the sequence--but shouldn't have happened in this case because the lukE gene you sent only has a stop codon at the end. Also, .gamma file you sent lists the strand as "-", which I did not get either. Is this the correct contig file?

Thanks--Rich

**Comment by BertBog** [comment]
Hey Rich,

Thanks for your reply! You are right, I used the reverse complement of the *lukE.fasta* in my analysis which caused the difference in the output. Unfortunately many sequences in database that I use are not standardized from 5' -> 3', so I'll have to fix that. 
I re-ran it with the sequences I uploaded here previously and I get the exact same result as you.
My apologies for the mistake.

Best,
Bert
